account-kit/rn-signer/android/src/main/java/com/accountkit/reactnativesigner/core/TEKStamper.kt

```diff
@@ -196,4 +201,71 @@ class TEKStamper(context: Context) {
             )
         return Pair(compressedPublicKey, privateKey)
     }
+
+    private fun createSharedPreferences(masterKey: MasterKey, context: Context): SharedPreferences {
+        return EncryptedSharedPreferences.create(
+            context,
+            ENCRYPTED_SHARED_PREFERENCES_FILENAME,
+            masterKey,
+            EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+            EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+        )
+    }
+
+    private fun createMasterKey(context: Context): MasterKey {
+        try {
+         return MasterKey.Builder(context.applicationContext, MASTER_KEY_ALIAS)
+            .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
+            // requires that the phone be unlocked
+            .setUserAuthenticationRequired(false)
+            .build()
+        } catch(createMasterKeyException: Exception) {
+            createMasterKeyException.printStackTrace()
+            throw RuntimeException("Error creating master key")
+        }
+    }
+
+
+    private fun getSharedPreferences(context: Context): SharedPreferences {
+        var encryptedSharedPreferences: SharedPreferences
+        try {
+            // Attempt to create or load the EncryptedSharedPreferences file
+            val masterKey = createMasterKey(context)
+            encryptedSharedPreferences = createSharedPreferences(masterKey, context)
+        } catch(e: Exception) {
+            // Log the Exception
+            e.printStackTrace()
+
+            // Delete the existing master key and EncryptedSharedPreferences
+            try {
```

Comment:
I find the nested try-catches a bit hard to read and possibly unnecessary. I might remove this outer one and just have tigher try-catches around the inner fallible operations.